util/cache/cache.go

```diff
@@ -33,20 +33,31 @@ func NewCache(client CacheClient) *Cache {
 	return &Cache{client}
 }
 
-func buildRedisClient(redisAddress, password, username string, redisDB, maxRetries int, tlsConfig *tls.Config) *redis.Client {
-	opts := &redis.Options{
-		Addr:       redisAddress,
-		Password:   password,
-		DB:         redisDB,
-		MaxRetries: maxRetries,
-		TLSConfig:  tlsConfig,
-		Username:   username,
+func buildRedisClient(redisAddress, password, username string, redisDB, maxRetries int, tlsConfig *tls.Config, redisClusterMode bool) *redis.Client {
+	var client *redis.Client
+	if redisClusterMode {
+		opts := &redis.ClusterOptions{
```

Comment:
share the same opts

Reply:
not sure if both sentinel config and cluster redis config can work together, as far as I understand, it's either a simple client, a cluster client, or a sentinel client.

Reply:
Share the same configuration based on parameter switching

Reply:
updated